src/main/java/frc/robot/arm/SimArmIO.java

```diff
@@ -0,0 +1,54 @@
+package frc.robot.arm;
+
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import frc.robot.Constants;
+
+public class SimArmIO implements IArmIO{
+    private static final double TICK = 2.0 * Math.PI / Constants.Arm.ENCODER_COUNT;
+
+    private final SingleJointedArmSim armSim = new SingleJointedArmSim(
+        DCMotor.getNEO(2),
+        100,
+        SingleJointedArmSim.estimateMOI(Units.inchesToMeters(30), 13), 
+        Units.inchesToMeters(30), 
+        Units.degreesToRadians(Constants.Arm.HOME_POSITION), 
+        Units.degreesToRadians(200), 
+        true, 
+        Units.degreesToRadians(Constants.Arm.HOME_POSITION), 
+        VecBuilder.fill(TICK)
+    );
+
+    private double angleRads;
+    private double motorSpeed;
+    private DigitalInput homeSwitch;
```

Comment:
This is a software-only implementation, so we don't need a DigitalInput. THis can just be a boolean that is stored and communicated back when `atFrontLimit` is called. 

How should that boolean change? Can you come up with some ideas? We can discuss more on Saturday if you don't come up with an idea.